src/sqlBible/src/SqlBible.ts

```diff
@@ -0,0 +1,292 @@
+        const section = new BibleSection({
+            phraseStartId: newPhrases[0].id!,
+            phraseEndId: newPhrases[newPhrases.length - 1].id!,
+            level: 0
+        });
+        return getManager().save(section);
+    }
+
+    async addPhrase(phrase: BiblePhrase) {
+        await this.dbReady;
+        await this.preparePhraseForDb(phrase);
+        return await getManager().save(phrase);
+    }
+
+    async addPhrases(phrases: BiblePhrase[]) {
+        await this.dbReady;
+        let verse,
+            phraseNum = 0;
+        for (const phrase of phrases) {
+            await this.normalizePhrase(phrase);
+            if (!verse || verse != phrase.normalizedVerseNum) {
+                verse = phrase.normalizedVerseNum;
+                phraseNum = await this.getNextPhraseIdForNormalizedVerseNum(
+                    phrase,
+                    phrase.versionId
+                );
+            } else {
+                phraseNum++;
+            }
+            phrase.phraseNum = phraseNum;
+        }
+
+        return getManager().save(phrases);
+    }
+
+    async addSection(section: BibleSection) {
+        await this.dbReady;
+        return await getManager().save(section);
+    }
+
+    async addVersion(version: BibleVersion) {
+        await this.dbReady;
+        return getManager().save(version);
+    }
+
+    async createCrossReference(refRange: IBibleReferenceRangeVersion) {
+        return new BibleCrossReference(this.getNormalizedReference(refRange));
+    }
+
+    async generateBookMetadata(book: BibleBook) {
+        await this.dbReady;
+        const metaData = await getManager()
+            .createQueryBuilder(BiblePhrase, 'phrase')
+            .addSelect('COUNT(DISTINCT phrase.versionVerseNum)', 'numVerses')
+            .where({
+                id: Between(
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 1,
+                            normalizedVerseNum: 1,
+                            versionId: book.versionId
+                        },
+                        5
+                    ),
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 999,
+                            normalizedVerseNum: 999,
+                            versionId: book.versionId
+                        },
+                        5
+                    )
+                )
+            })
+            .orderBy('phrase.versionChapterNum')
+            .groupBy('phrase.versionChapterNum')
+            .getRawMany();
+        book.setChaptersMeta(metaData.map(chapterMetaDb => chapterMetaDb.numVerses));
+        return getManager().save(book);
+        console.log(metaData);
+    }
+
+    async getBookForVersionReference(reference: IBibleReferenceVersion) {
```

Comment:
you could possibly use object destructuring in the function parameter, to get `versionId` and `bookOsisId`- not strictly necessary but nice to have

Reply:
good. done.